src/basic/calendarspec.h

```diff
@@ -36,6 +36,7 @@ typedef struct CalendarComponent {
 
 typedef struct CalendarSpec {
         int weekdays_bits;
+        int day_offset;
```

Comment:
please us our usual "usec_t" type here. It's what parse_time() uses as output, and it is what format_timespan() will take as input. Now, usec_t is unsigned, so I'd suggest adding two variables here: "usec_t offset" and "bool offset_negative", the latter indicating whether to add or subtract the offset value.